File: core/core-api/src/main/java/com/coffee/gu/api/controller/v1/response/order/OrderResponse.java
package com.coffee.gu.api.controller.v1.response.order;

import com.coffee.gu.order.Order;
import com.coffee.gu.enums.OrderState;

import java.math.BigDecimal;
import java.util.List;

public record OrderResponse(
        String key,
        String name,
        BigDecimal totalPrice,
        OrderState state,
        List<OrderLineResponse> lines
) {
    public static OrderResponse from(Order order) {
        return new OrderResponse(
                order.getKey(),
                order.getName(),
                order.getTotalPrice(),
                order.getState(),
                order.getLines()
                        .stream()
                        .map(line -> new OrderLineResponse(
                                line.getMenuId(),
                                line.getMenuName(),
                                line.getImageUrl(),
                                line.getDescription(),
                                line.getQuantity(),
                                line.getUnitPrice(),
                                line.getTotalPrice()
                        ))
                        .toList()
        );
    }
}

File: core/core-api/src/main/kotlin/com/coffee/gu/api/controller/v1/response/order/OrderResponse.kt
package com.coffee.gu.api.controller.v1.response.order

import com.coffee.gu.enums.OrderState
import com.coffee.gu.order.Order
import java.math.BigDecimal

class OrderResponse(
    val key: String,
    val name: String,
    val totalPrice: BigDecimal,
    val state: OrderState,
    val lines: List<OrderLineResponse>,
) {
    companion object {
        fun from(order: Order): OrderResponse {
            return OrderResponse(
                key = order.key,
                name = order.name,
                totalPrice = order.totalPrice,
                state = order.state,
                lines = order.lines.map { line ->
                    OrderLineResponse(
                        menuId = line.menuId,
                        menuName = line.menuName,
                        imageUrl = line.imageUrl ?: "",
                        description = line.description ?: "",
                        quantity = line.quantity,
                        unitPrice = line.unitPrice,
                        totalPrice = line.totalPrice
                    )
                }
            )
        }
    }
}
